src/site/content/sitemap.xml.njk

```diff
@@ -6,8 +6,11 @@ permalink: "/sitemap.xml"
 <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
 {%- for page in collections.all %}
 {%- if not page.data.noindex and not page.data.draft %}
+  {# Verify that the page has a permalink so we don't pull in things like #}
+  {# the service worker partial. #}
+  {# But use the page's fileSlug so we can strip out the language directory. #}
   {%- if page.url %}
-  {% set absoluteUrl %}{{ page.url | absoluteUrl(site.url) }}{% endset %}
+  {% set absoluteUrl %}{{ page.fileSlug | absoluteUrl(site.url) }}{% endset %}
```

Comment:
FileSlug is not reliable. E.g. for the file /newsletters/archive/2020/03, slug is just "03", which does not give you a correct url for this resource. in PR #3261 I introduced canonicalUrl to address that. Maybe use canonicalUrl instead?